pkg/cloud/cloud.go

```diff
@@ -1275,19 +1276,28 @@ func (c *cloud) CreateSnapshot(ctx context.Context, volumeID string, snapshotOpt
 	descriptions := "Created by AWS EBS CSI driver for volume " + volumeID
 
 	var tags []types.Tag
+	var request *ec2.CreateSnapshotInput
 	for key, value := range snapshotOptions.Tags {
 		tags = append(tags, types.Tag{Key: aws.String(key), Value: aws.String(value)})
 	}
 	tagSpec := types.TagSpecification{
 		ResourceType: types.ResourceTypeSnapshot,
 		Tags:         tags,
 	}
-	request := &ec2.CreateSnapshotInput{
-		VolumeId:          aws.String(volumeID),
-		TagSpecifications: []types.TagSpecification{tagSpec},
-		Description:       aws.String(descriptions),
+	if snapshotOptions.OutpostArn != "" {
+		request = &ec2.CreateSnapshotInput{
+			VolumeId:          aws.String(volumeID),
+			TagSpecifications: []types.TagSpecification{tagSpec},
+			Description:       aws.String(descriptions),
+			OutpostArn:        aws.String(snapshotOptions.OutpostArn),
```

Comment:
Can you check this? I'm assuming in the case when users don't provide `outpostArn`, we end up setting this to `""` instead of `null` so `CreateSnapshot` would fail.

Reply:
In the case there is no outpostArn parameter in the VSC than the case in the switch statement in controller.go is never hit so the outpost arn passed to cloud.go is the empty string "" in this case we do not want to give the ec2 api an outpostarn so we drop to the else. Ether way we are creating a createSnapshotInput and passing it to ec2.

Reply:
Ahh, my bad. In that case, can we just create a single request object, and do the check later?

```
req := &ec2.CreateSnapshotInput {
   ...
}

if options.OutpostArn != "" {
       req.OutpostArn = aws.String(options.OutpostArn)
}
```

Reply:
Ahhh yes we can that. Will add to next rev thank you !